### PR TITLE
EVG-6962: make host allocator more conservative with dep queue

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -16,14 +16,17 @@ import (
...
+	ID                   string
+	IsSecondaryQueue     bool
+	IncludesDependencies bool
+	StartedAt            time.Time
 }
 
 type TaskPlanner func(*distro.Distro, []task.Task, TaskPlannerOptions) ([]task.Task, error)
 
 func PrioritizeTasks(d *distro.Distro, tasks []task.Task, opts TaskPlannerOptions) ([]task.Task, error) {
+	opts.IncludesDependencies = d.DispatcherSettings.Version != evergreen.DispatcherVersionRevisedWithDependencies
```
Comment (bsamek) [inline scheduler/scheduler.go]: This means that `opts.IncludeDependencies` will be false if the dispatcher version isn't `DispatcherVersionRevisedWithDependencies`. That seems wrong.

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -149,17 +157,28 @@ func GetDistroQueueInfo(distroID string, tasks []task.Task, maxDurationThreshold
 		}
 
 		duration := task.FetchExpectedDuration().Average
-		task.ExpectedDuration = duration
-		distroExpectedDuration += duration
+
+		if !opts.IncludesDependencies {
+			task.ExpectedDuration = duration
+			distroExpectedDuration += duration
+		} else if depsMet, _ := task.DependenciesMet(depCache); depsMet {
```
Comment (bsamek) [inline scheduler/scheduler.go]: Make this an `||` so that you don't have to duplicate these lines?

File: scheduler/scheduler.go (hunk 1 above)
Comment (tychoish) [reply scheduler/scheduler.go]: that is exactly correct

File: scheduler/scheduler.go (hunk 2 above)
Comment (tychoish) [reply scheduler/scheduler.go]: yeah.... the assignment in the if statement doesn't seem to compile if its in an || statement.

File: scheduler/scheduler.go (hunk 1 above)
Comment (bsamek) [reply scheduler/scheduler.go]: But "with dependencies" implies "includes dependencies".

Comment (tychoish) [reply scheduler/scheduler.go]: every time I read you comment it still looks right/makes sense to me.

Comment (bsamek) [reply scheduler/scheduler.go]: oh

**Pull request merged**